concept/type/impl/AttributeTypeImpl.java

```diff
@@ -139,6 +142,10 @@ public void setSupertype(AttributeType superType) {
         } else if (!superType.isAbstract()) {
             throw exception(TypeDBException.of(ATTRIBUTE_NEW_SUPERTYPE_NOT_ABSTRACT, superType.getLabel()));
         }
+        SchemaValidation.throwIfNonEmpty(Iterators.link(
```

Comment:
Let's split the stuff above here into `validateDeclaration`-type methods, so we have some symmetry with `SubtypeValidation`.